### ISSUE TITLE
Scripts like `prettify` are sorted in wrong order

### ISSUE THREAD

**Issue by brc-dd** [cameronhunter/prettier-package-json#44]
If I have a JSON file like this:

```json
{
  "scripts": {
    "prepare": "echo prepare",
    "preprettify": "echo preprettify",
    "prettify": "echo prettify",
    "postprettify": "echo postprettify",
    "prestart": "echo prestart",
    "start": "echo start",
    "watch": "echo watch"
  }
}
```

Then the tool sorts this to:

```json
{
  "scripts": {
    "prepare": "echo prepare",
    "preprettify": "echo preprettify",
    "postprettify": "echo postprettify",
    "prestart": "echo prestart",
    "start": "echo start",
    "prettify": "echo prettify",
    "watch": "echo watch"
  }
}
```

Note the position of `prettify`. It is treating `prettify` as the `pre` script of `ttify`. Your smart sorting logic should be modified to check if `ttify` is there as a key, then only consider `prettify` as a `pre` script.

The fix can be approached like this: <https://github.com/brc-dd/prettier-package-json/blob/main/src/sort-scripts.ts>. The project's coding style is different that what I did there, that's why I am not creating a PR.

**Comment by cameronhunter** [comment]
Ha! That's a good bug. I won't have time to address this but PRs are welcome if you'd like to tackle it yourself.

**Comment by brc-dd** [comment]
@cameronhunter Actually I have linked a fix above can you check that if it's fine or not? Also, one of your tests is failing on your `main` branch itself. Here are the logs:

```txt
➜ yarn test
yarn run v1.22.5
$ jest
 PASS  tests/script-order.test.ts
 PASS  tests/use-tabs.test.ts
 PASS  tests/tab-width.test.ts
 PASS  tests/sort-contributors.test.ts
 PASS  tests/sort-files.test.ts
 PASS  tests/key-order.test.ts
 FAIL  tests/command-line.test.ts (6.573 s)
  ● prettier-package-json --list-different tests\__fixtures__\invalid.json

    expect(received).toMatch(expected)

    Expected pattern: /tests\/__fixtures__\/invalid\.json: Unexpected token t in JSON at position 6/
    Received string:  "SyntaxError: tests/__fixtures__/invalid.json: Unexpected token t in JSON at position 7
        at JSON.parse (<anonymous>)
        at Object._readFile (C:\\Users\\brc-dd\\Desktop\\prettier-package-json\\node_modules\\jsonfile\\index.js:25:16)
        at async check (C:\\Users\\brc-dd\\Desktop\\prettier-package-json\\bin\\prettier-package-json:91:18)
        at async Promise.all (index 0)
    "

      29 |   expect(result.exitCode).toBe(1);
      30 |   expect(result.stdout).toBe('');
    > 31 |   expect(result.stderr).toMatch(/tests\/__fixtures__\/invalid\.json: Unexpected token t in JSON at position 6/);
         |                         ^
      32 | });
      33 |

      at tests/command-line.test.ts:31:25

 › 7 snapshots written.
 › 7 snapshots obsolete.
   • prettier-package-json --config tests/__fixtures__/prettier-package-json.config.js tests/__fixtures__/package-1.json 1
   • prettier-package-json --list-different tests/__fixtures__/missing.json 1
   • prettier-package-json --list-different tests/__fixtures__/package-*.json 1
   • prettier-package-json --tab-width 8 tests/__fixtures__/package-1.json 1
   • prettier-package-json --use-tabs tests/__fixtures__/package-1.json 1
   • prettier-package-json tests/__fixtures__/package-*.json 1
   • prettier-package-json tests/__fixtures__/package-1.json 1
Snapshot Summary
 › 7 snapshots written from 1 test suite.
 › 7 snapshots obsolete from 1 test suite. To remove them all, run `yarn test -u`.
   ↳ tests/command-line.test.ts
       • prettier-package-json --config tests/__fixtures__/prettier-package-json.config.js tests/__fixtures__/package-1.json 1
       • prettier-package-json --list-different tests/__fixtures__/missing.json 1
       • prettier-package-json --list-different tests/__fixtures__/package-*.json 1
       • prettier-package-json --tab-width 8 tests/__fixtures__/package-1.json 1
       • prettier-package-json --use-tabs tests/__fixtures__/package-1.json 1
       • prettier-package-json tests/__fixtures__/package-*.json 1
       • prettier-package-json tests/__fixtures__/package-1.json 1

Test Suites: 1 failed, 6 passed, 7 total
Tests:       1 failed, 24 passed, 25 total
Snapshots:   7 obsolete, 7 written, 17 passed, 24 total
Time:        19.69 s, estimated 20 s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by JamesGelok** [comment]
@brc-dd, I'm not trying to speak on behalf of anyone but myself _but_ I wonder if it would be more useful to make a PR based on the changes in your fork? We don't know how busy `cameronhunter` is and it's easier (imo) to verify your changes against the current version in a PR.

That being said, I don't wanna assume your situation either; you may have other changes on that branch you're working on. 

So, I made a diff using this website to observe the changes in `/src/sort-scripts.ts`:

https://www.diffchecker.com/ooH1CJqj

**Comment by brc-dd** [comment]
@JamesGelok Actually there are no other changes, its perfectly mergeable, but seeing this project style was quite different from mine, I haven't yet created a PR. And, perhaps you're right. It will be easier to review on PR. I will create one after adding tests for the change.

**Comment by JamesGelok** [comment]
@brc-dd Awesome! ^_^

And the stylistic difference was some friction I felt too when I was trying to get prettier-ignore to work. 

I made [this PR](https://github.com/cameronhunter/prettier-package-json/pull/43) a while back after I noticed that the prettier style is – for the most part – the same as the one in [the config](https://github.com/cameronhunter/ink-markdown/blob/f97fb5cf1534d64c86e2ed3d412abacdb45ded6c/.prettierrc.js) of [this other project](https://github.com/cameronhunter/ink-markdown) @cameronhunter was working on.

<hr/>

FWIW, I [ran your code through prettier](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgGdcYB5AIwCs4wCAzHCdfAcmLwFoIqaYOBrOAE9CLANwAdKBmx58uACZwcAISH4GTVuz6Ll4qTJL5g+AAoBDMPwsBzOACliUfAF8NjZiwB0AehhCmHCiklBSkFCEBGAAFjT8Zjhw+AC8+AAUFji2SEQwOKhQtgA0RGAFmDCEuZbWdo7OANps5aiVogC6AJS55BAQADZwFi4pAHxS+Pi+AHqYSb7e8FGZ2V34AGQbZRVVAITeMRaEpADuUIkQQXhCq7behACu5FE46QDMXV2h4dBR+LF4mYIP80ndcq9CiUdm0qjUrDZ7E5oM1CK12ixur1+kMRqkJi5pnMQTBFssYHd1lsYe0DkcTudLtcAncHs9XukACxfH5QXy+fAAZWMFgGmCO5DgMFQYFFA3U5HUaN2+CgFnQyTgAA8wANHgoofh5nBfNh-sa6KgtcEpHRHlBaKhoDSYAA5dVwdIwGKoarmBH1ZFQVHoqqY0reSNZWx+xqQoqleO2LH4ACM+AAPvgAAyZ-AcdPASYAv4ERoWUrkDqpfDRwjedAWTDpSn44tTQHWRKe6Olb2+9YZrOdhIku59n2EdYAflr2W8SUwAysntm6WNGbNMC6vlKLBY61yveL32LSRgjxwLgsqRSaXI+FnuaP+AAPPgH7OC-hcqnQq4pCkfkhWMRUjSSRNQ1KLdbXtR1nWNYEogACX6fhCC9Sd4TqJEmhaXZOlKCwIXyKFKxIgoih6NM81zLNvyLQlUDoDIR27TIJwHTNhziLsx3ITip3Wc9LxcAtQimOABkIZJmNY3jRxWCt8H7KduIBBT2IElTJy+fARKvNMJP0qVRJzf9AKgbVZAIJQ6AsR4BnoODpWdHRBVDDDlVhP1akRBoUXwnzMRrYBXGoxipm89pp2w-ygxDAjMX-Exiwif49BwGtMrUdJordD1vHIQoFDyzyumgpIkJgVCIHQoqSrKgieTS0t8EEEQawoahaG8DqvPK2tCFVOAADdlEaDpjIMlx+u8IYim9W80lzWcwp-EwXT9HRut4JqfNKOadHSTK9NcCywj5AUIktWwNFQIZckKfgOEbHB+AUCBzkWY0YGlZQcDAbxKEIICBQGZ7chiP7MGqflbFQb1nm8SB0F8WUNUYKAYnteAcF8Z7XqyD6vr5cgBm4Xw6AATgAdjocgAFYwDoVNGfeTkFAANk5MAAA4ubgAAmOAFHeBROVTIWLHIKwFHITlGaUbmwB+89-pwQHgdBq7VQgeAkDBqZjYAAwgcbNdQJRCBN9rhHwX09YIJI6CGWhRYdlxCCtuBZZwYojeNj9HgIRHhqgfWRtFj26FwHTHctIYPzAB2qid-ATZNhsFCzsGjfQCAFEcuBvGskhhrSRjgKmeZChgAB1K3vV-IXswD3WphgGXG4UZv8CF9vq-wR4ZIAFRlv17OkuBB4FKK4HQVBcnyR4Z8DogoSGABFR59bgZecFX2fjZBrVBU3uAd73g+j-X-ILAeqEAGEmEbXIWAjqyWGPqZyBwBEpSCkwFYKEN814d3wKfFQ-9rCAI9AAGUKPvDQooZI-znIwU4lgkiRHfqKU4FgRDf3XvMEEcB67-0wHggYBCiHoPNgDH2sZ15TCrnPIO90hh+hYAAKmzsQiBQcriuVwalQRHCu7kB7n3AeLCOG11gNImIuQ+bZjkVMAC4jNFDw6Ebc6UgQDFBAMIp0kRkCgCyJg7BCBCDIBAPgwhtijF-wATAIBICijIBXjPEAC9JQKGVvAkYthHj1AAGK4EbH9KEdiHIwAgIYkA0N0ADHrj6ZYwCwBwEFDYxGqBRqIyEHYsAhAnFoEiMoGAiQ7CNmQFPNBIAoEwMEG4hBSC6moJ8afc+RRt673gF4w+PjgE4BkjgOxstJQDA4FURJCiG5NxiMgKWbcQCkJkhQpssSaGOMSQAR36XAJktiUDHA4FZaOChElJAOagKq-9bC1KQPUnxMlF6DKPiAb2vTL6HI+T4yRSjkADxAPfR+RQX7oCeSAT+cBEkjzgOPF4HTp5GNGqvAAklAJQsAPK7AAILYsFAEIYKKZKuFcEAA) using the config from that other project.

And except for a single comma, you matched the style _**perfectly**_. ([diff](https://www.diffchecker.com/WwXRqBui))

**Comment by cameronhunter** [comment]
I lost track of this issue, but I think it's fixed by the recently released version `v2.8.0` (via #66).
